NumberTheory/FastPower.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+<Fast Power>
+--------------
+Fast Power is an optimized algorithm to compute exponentiation in a short time.
+
+
+Time Complexity
+-----------------
+O(log(N)) where N is the power the number is raised to.
+
+Space Complexity
+------------------
+O(log(N)) where N is the power the number is raised to.
+
+*/
+#include <iostream>
+
+using namespace std;
+typedef long long ll;
+
+//Function that returns x raised to the power of y
+ll fastPower (ll x,ll y)
```

Comment:
Please use more meaningful variable names like `base` and `exponent` instead of just `x` and `y`.
Also, rename the function to `fast_power` to adhere to the guidelines.